niceplots/utils.py

```diff
@@ -1,127 +1,109 @@
-    niceColors["Orange"] = "#ff8f00"
-    niceColors["Purple"] = "#800080ff"
-    niceColors["Cyan"] = "#00A6D6"
-    niceColors["Black"] = "#000000ff"
-    # The 2 colours below are not used in the colour cycle as they are too close to the other colours.
-    # Grey is kept in the dictionary as it is used as the default axis/tick colour.
-    # RedOrange is the old Orange, and is kept because I think it looks nice.
-    niceColors["Grey"] = "#5a5758ff"
-    niceColors["RedOrange"] = "#E21A1A"
-
-    return niceColors
-
-
-def get_delftColors():
-    # Define an ordered dictionary of the official TU Delft colors to use as the default color cycle
-    delftColors = OrderedDict()
-    delftColors["Cyan"] = "#00A6D6"  # '#f8a30dff'
-    delftColors["Yellow"] = "#E1C400"
-    delftColors["Purple"] = "#6D177F"
-    delftColors["Red"] = "#E21A1A"
-    delftColors["Green"] = "#A5CA1A"
-    delftColors["Blue"] = "#1D1C73"
-    delftColors["Orange"] = "#E64616"
-    delftColors["Grey"] = "#5a5758ff"
-    delftColors["Black"] = "#000000ff"
-
-    return delftColors
+    # If the style is a niceplots style, return the file path
+    if styleName in get_available_styles():
+        curDir = os.path.dirname(os.path.abspath(__file__))
+        return os.path.join(curDir, "styles", styleName + ".mplstyle")
+
+    # Otherwise assume it's a matplotlib style and just return the style name
+    return styleName
+
+
+def get_colors():
+    """
+    Get a dictionary with the colors for the current style. This function
+    only works when niceplots styles are used (not built-in matplotlib ones).
+
+    Returns
+    -------
+    dict
+        Dictionary of the colors for the requested style. The keys
+        are human-readable names and the keys are the hex codes.
+    """
+    # Get the color codes and their names from the (hopefully) "special" parameter
+    color_codes = get_colors_list()
+    color_names = plt.rcParams["keymap.help"]
+
+    # Ensure that the amount of color names matches the amount of colors
+    assert len(color_codes) == len(
+        color_names
+    ), "The colors are not properly named in the stylesheet, please open an issue on GitHub with the details!"
+
+    # Make the dictionary and return it
+    return dict(zip(color_names, color_codes))
+
+
+def get_colors_list():
+    """
+    Get a list with the colors for the current style. This function
+    works with all matplotlib styles.
+
+    Returns
+    -------
+    list
+        List of the colors for the requested style.
+    """
+    return plt.rcParams["axes.prop_cycle"].by_key()["color"]
+
+
+def get_available_styles():
+    """
+    Get a list of the names of styles available.
+
+    Returns
+    -------
+    list
+        The names of the available styles.
+    """
+    curDir = os.path.dirname(os.path.abspath(__file__))
+    styleFilenames = os.listdir(os.path.join(curDir, "styles"))
+    styles = []
+    for s in styleFilenames:
+        if s[-9:] == ".mplstyle":
```

Comment:
A better way is to use [this function](https://docs.python.org/3/library/os.path.html#os.path.splitext). Also, why do we need this check? Does `styleFilenames` sometimes include the extension and sometimes omit it?

Reply:
It should always include it since niceplots handles adding the extension when the style is requested. I changed it so it enforces that here.